packages/identity-dialect-dapps/src/identity-dialect-dapps.ts

```diff
@@ -0,0 +1,45 @@
+import type {
+  Identity,
+  IdentityResolver,
+  ReverseIdentity,
+} from '@dialectlabs/sdk';
+import { PublicKey } from '@solana/web3.js';
+import { Dapp, fetchAllDapps } from './api';
+
+const NAME = 'DIALECT_DAPPS';
+
+export class DialectDappsIdentityResolver implements IdentityResolver {
+  // public key to identity
+  dappsCached?: Record<string, Identity>;
```

Comment:
Should we periodically re-cache this?

Reply:
I don't think so, dapps are rarely to change, and in the frontend use-case page refresh will re-cache this basically